### DISCUSSION TITLE
Can timeVal() be safely used for a door buzzer?

### DISCUSSION THREAD

**Discussion by etkaar** [HomeSpan/HomeSpan#809] [General]
I have a door buzzer which needs to be retained as long one wants to open the door, so it is not a clicky one.

Instead of using `HIGH > delay(2000) > LOW` within `update()` e.g. for retaining it for two seconds, I release it within `loop()` (if still retained) by checking `getVal()` and `timeVal()`.

My question now is: Is this a safe approach or do I risk a failure in case the connection to HomeKit is lost or so?

```
boolean update() {
	if (buzzer->getNewVal() == 1) {
		digitalWrite(this->pin_number, HIGH);
		WEBLOG("Retain door buzzer for %d ms (GPIO%d): %d", DOOR_BUZZER_RETAIN_PERIOD, this->pin_number, buzzer->getNewVal());
	} else {
		digitalWrite(this->pin_number, LOW);
		WEBLOG("Release door buzzer.");
	}

	return(true);
}

void loop() {
	if (buzzer->getVal() && buzzer->timeVal() > DOOR_BUZZER_RETAIN_PERIOD) {
		digitalWrite(this->pin_number, LOW);
		WEBLOG("Automatically release door buzzer.");

		buzzer->setVal(0, true);
	}
}
```

**Comment by HomeSpan** [comment]
Is the door buzzer the doorbell?  Or do you mean that the door buzzer is the mechanism that actually unlocks the door?

If the latter, see the EmulatedPushButton tutorial - I think it may have what you are looking for.

**Reply by etkaar** [reply]
I think 'electric strike' would be the right word for it. But regarding my other question: Is that safe? What happens if the connection is lost within the timeframe to Home Kit? Is the timeout from `getVal()` and `timeVal()` in order to stop the electric strike guaranteed then?

**Reply by HomeSpan** [reply]
Once you trigger from HomeKit, all the timing is done within the ESP32.  If you lose connection to HomeKit the only thing that would happen is the button icon would still show as pressed in the Home App, even though the device already released it.

**Reply by etkaar** [reply]
Okay, so definitely no `HIGH > delay(2000) > LOW` required and I can do it as I have shown in my example in `loop()`, right? Thank you!
